### DISCUSSION TITLE
Visual studio 2022 diagnostic tools not working for aspire projects

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#4065] [Q&A]
Hi there, recently I'm not seeing the profiling diagnostic tools working when debugging my Aspire app (still working for non-aspire apps).

This is on Version 17.10.0 Preview 6.0 with Aspire preview 6.

It says: "the diagnostic tools window does not support the current debug configuration". Is this just me, or more widespread?

![image](https://github.com/dotnet/aspire/assets/2116108/7f99509c-a0ed-4c38-a8a8-754fba974d42)

**Comment by karpinsn** [comment]
Nope, this is not just you. We intentionally disabled the tools for Aspire projects for now since the tools only support single process debugging. We are investigating how to enable multiprocess debugging in the tools which would make them applicable to Aspire projects. This is being tracked with the following [Visual Studio Suggestion Ticket](https://developercommunity.visualstudio.com/t/support-multiple-process-profiling/423528). I realize the ticket is a bit old but we are actively investigating this now to hopefully support .NET Aspire.

**Comment by mip1983** [comment]
Thanks @karpinsn. Is it possible in the interim to still have single process debugging enabled and be able to choose a single process?

**Comment by karpinsn** [comment]
> Is it possible in the interim to still have single process debugging enabled and be able to choose a single process?

Not yet, though we are working towards that. There are two things to work through here, the first is collecting the data from the different processes, the second is figuring out the multi-process user experience. We are starting with collecting from multi-processes and using a process selector, then we are going to figure out how to properly show multiple processes in the user experience. 

Right now if we enable the tools for the scenario we show data from whatever is the first process we get the process start event from which is random. This felt weird and broken so we opted to disable it until we can provide a more reasonable experience.

**Comment by Suundumused** [comment]
may 2025 and still not working...

**Comment by sagarok** [comment]
August 2025.. The same problem. :(

**Comment by akrotov** [comment]
Same problem. Added the ability for one of aspire services (which needs profiling) to run without rest of cluster (configured it to use external DB and it has no other dependencies on cluster services). Tried to profile service as regular .NET service, selected it as startup in visual studio with debug profile with needed settings for standalone mode and instantly receive: Microsoft Visual Studio canceled starting your diagnostics session.

In diagnostic hub there is a message:
Cannot enable stacks for specific custom event ID(s) if stacks were enabled for all custom event IDs.


I thought that it is a bug. But reading reagrding disabling I have a question. Is my assumption correct, that due to incorrect work of profiler with multiprocess aspire envrionment, you completely forbidden profiling .NET services, which are part of aspire solution. making it impossible to profile the service even in standalone mode (where there should be no problems, as it is regular .NET service if it is launched outside the aspire)? Sounds too wired, that's why I am asking.

If yes - are there any workarounds which you haven't disabled yet? :-)

**Comment by karpinsn** [comment]
Hey @akrotov, are you using the Diagnostic Tools while debugging or the Performance Profiler? We added multi-proc support for Aspire into the Performance Profiler, currently for the CPU Usage Tool and Instrumentation Tool. When you select the tool you need to select the "Collect data from multiple processes" checkbox and it will automatically attach to each Aspire process:
<img width="1001" height="671" alt="image" src="https://github.com/user-attachments/assets/215ce9ed-64ca-4502-b8b4-00a2caf15008" />

Let me know if this isn't working for you and we can investigate.

**Reply by jimSampica** [reply]
Works good. Looking forward to the Memory profiler 😁

**Reply by karpinsn** [reply]
@jimSampica are you looking for the Memory Usage tool or the .NET Object Allocation tool?

**Reply by jimSampica** [reply]
Memory usage tool
